crates/release_plz_core/src/release.rs

```diff
@@ -131,9 +140,41 @@ fn release_package(
         info!("published {}", package.name);
     }
 
+    if input.gh_release {
+        publish_release(git_tag(&package.name, &package.version.to_string()), input);
+    }
+
     Ok(())
 }
 
+async fn publish_release(git_tag: String, input: &ReleaseRequest) {
+    let repo_url = match &input.repo_url {
+        Some(url) => RepoUrl::new(url.as_str()),
+        None => {
+            let repo = Repo::new(input.workspace_root().unwrap()).unwrap();
```

Comment:
We already have created the `repo` on line 120.
We could move this piece of code in that function and reuse that variable, instead of creating it here

Reply:
Also, we could extract these lines because they are repeated [here](https://github.com/MarcoIeni/release-plz/blob/f03c598c38e8cf5ceeae542add5d54fae3915159/crates/release_plz/src/args/update.rs#L92-L94).

I wrote this function you can add [here](https://github.com/MarcoIeni/release-plz/blob/f03c598c38e8cf5ceeae542add5d54fae3915159/crates/release_plz_core/src/repo_url.rs#L34):

```rust
    pub fn from_repo(repo: &Repo) -> Result<Self, anyhow::Error> {
        let url = repo.origin_url().context("cannot determine origin url")?;
        RepoUrl::new(&url)
    }
```